src/System.Text.Json/src/System/Text/Json/CustomUncheckedBitArray.cs

```diff
@@ -44,6 +57,69 @@ public int MaxIndexableLength
             }
         }
 
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public void EnsureAndPushTrue(int depth)
```

Comment:
Nit: I like "PushTrue", but the "At" confuses me a little simply because the depth < AllocationFreeMaxDepth path ignores the depth and always pushes on the top of the stack.

Reply:
If we keep "At", there is confusion for depth < AllocationFreeMaxDepth. If we remove it, there is confusion as to why the API is accepting a depth and not always pushing to the top. Since the depth is relevant, I am leaning towards keeping it as PushTrueAt.